### ISSUE TITLE
Exiting with exit code 1, no warnings, no errors when executed on Github Action

### ISSUE THREAD

**Issue by fbramato** [doxygen/doxygen#10866]
**Describe the bug**
Doxygen is executed and exits immediately, no warnings, no errors.

**Screenshots**

![image](https://github.com/doxygen/doxygen/assets/118736163/e2abad7b-d65d-419e-92b7-1967495e9dfa)

**To Reproduce**
I'm running a Github Action Workflow

```
# Simple workflow for deploying static content to GitHub Pages
name: Deploy static content to Pages

on:
  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch: {}

# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
permissions:
  contents: read
  pages: write
  id-token: write

# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
concurrency:
  group: "pages"
  cancel-in-progress: false

jobs:
  # Single deploy job since we're just deploying
  build-and-deploy:
    runs-on: windows-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v4

      - name: Install Doxygen
        uses: ssciwr/doxygen-install@v1
        
      - name: Setup PHP with PECL extension
        uses: shivammathur/setup-php@v2
        with:
          php-version: "7.4"
          tools: composer

      - name: Setup Node
        uses: actions/setup-node@v4
    
      - name: NPM install
        run: npm install

      - name: Composer install
        run: composer install

      - name: Run Doxygen
        run: |
          doxygen bvi106php-r01BackEnd.Doxyfile
          doxygen bvi106vue-r01FrontEnd.Doxyfile
``` 

**Expected behavior**
Should run and creates the documentation.

**Version**
1.10
Tested on ubuntu-latest and windows-latest on Github Workflow, and via act (https://github.com/nektos/act) locally 

**Stack trace**
None

**Additional context**
Tested the same Doxyfile on Windows 11 and WSL, it works, generate correctly the documentation portal as expected.

**Comment by albert-github** [comment]
From the description it is not possible to see what the problem is. 
First steps are to show the `bvi106php-r01BackEnd.Doxyfile` and run again without `QUIET = YES` (I assume this is in the file `bvi106php-r01BackEnd.Doxyfile`) but with `QUIET = NO` so we have an idea where the problem lies (an publish this full result.
When the line `WARN_LOGFILE` is set to a file also disable this line.

What do you mean with:
> and via act (https://github.com/nektos/act) locally

as I don't see a `Doxyfile` or something similar.

**Comment by fbramato** [comment]
`act` is a tool for developing Github Workflow without pushing to remote, fundamentally executes the workflow locally using your docker as it would be excuted on GH, with some exceptions,

the Doxyfile was already with `QUIET = NO`.

But, after scratching my head for 2 days, i've found the reason:
`HTML_OUTPUT` was set to `./dist/backend` but `dist` folder was not created when the workflow is executed because gitignored.

Creating the `dist` folder it works and fires only a warning about not having `dist/backend` "but i will create it for you"

So seems the crash is due to not having the parent folder where the documentation should be created.

**Comment by albert-github** [comment]
So the background actually is that there is no warning on the console but this is probably in the `WARN_LOGFILE`.

I think this problem is related to the proposed pull request #8444

(as such reopening the issue).

**Comment by doxygen** [comment]
@albert-github There should be a warning of course when exiting with code 1. But creating whole paths recursively as suggested in #8444 I find rather dangerous. A small mistake (think of an environment variable used in the path which is not set correctly) might put the documentation somewhere it should not be, requiring the user to go clean up the mess.

**Comment by albert-github** [comment]
@doxygen 

The warning is present, but probably hidden in the file indicated by `WARN_LOGFILE` (I see the message in my console window when not using the setting or in the file when using the setting). @fbramato didn't confirm that he used `WARN_LOGFILE`.

It is definitely the case when automatically creating a tree of directories this can lead to have the directory at an unexpected place.
@doxygen so (please confirm so it can be done):
- we should close this issue
- close #8444 as not to be integrated with a reasoning like "A small mistake (think of an environment variable used in the path which is not set correctly) might put the documentation somewhere it should not be, requiring the user to go clean up the mess"

**Comment by albert-github** [comment]
Some more thoughts:

Maybe when having an `exit` in the middle of the doxygen program (like for the `OUTPUT_DIRECTORY` we should not use the constructs like:
```
        err("tag OUTPUT_DIRECTORY: Output directory '%s' does not "
            "exist and cannot be created\n",qPrint(outputDirectory));
        cleanUpDoxygen();
        exit(1);
```
but use
```
        term("tag OUTPUT_DIRECTORY: Output directory '%s' does not "
            "exist and cannot be created\n",qPrint(outputDirectory));
```
and extend the `term_` (in message.cpp) so the warning is not only written to the `WARN_LOGFILE` but when this is set also to `stderr`.

Another possibility would be to write to stderr (when`WARN_LOGFILE` is set something like "Please consult ... file for the reason".

**Comment by doxygen** [comment]
@albert-github I like the idea of using `term` instead of `exit` then always writing a message (which can be a reference to the log file).

**Comment by albert-github** [comment]
@doxygen 

Indeed I like the usage of `term` as well.
Will you implement it or should I do it?

**Comment by doxygen** [comment]
@albert-github you can do it

**Comment by albert-github** [comment]
Based on https://github.com/doxygen/doxygen/issues/10866#issuecomment-2118710447 and https://github.com/doxygen/doxygen/issues/10866#issuecomment-2118713980 I've just pushed a proposed patch, pull request #10876

Also seen the above #8444 is obsolete.

**Comment by fbramato** [comment]
> @doxygen
> 
> The warning is present, but probably hidden in the file indicated by `WARN_LOGFILE` (I see the message in my console window when not using the setting or in the file when using the setting). @fbramato didn't confirm that he used `WARN_LOGFILE`.

Sorry if i come across this only now: yes, we are using WARN_LOGFILE but in a Workflow Action, without having access to the log file or printing it, the developer should rely on correct console outputs.

**Comment by albert-github** [comment]
Maybe you should set in your settings:
```
WARN_AS_ERROR          = FAIL_ON_WARNINGS_PRINT
```
which is very suited for this type of usage.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
